netlogo-core/src/main/fileformat/PlotConverter.scala

```diff
@@ -5,36 +5,53 @@ package org.nlogo.fileformat
 import java.nio.file.Path
 
 import org.nlogo.core.{ CompilationEnvironment, Dialect, ExtensionManager,
-  LiteralParser, Model, Plot, Pen, SourceRewriter }
+  LiteralParser, Model, Plot, SourceRewriter }
 import org.nlogo.api.AutoConvertable
 
 object PlotConverter {
   def allPlotNames(model: Model): Seq[String] =
     model.plots.flatMap(_.display).filterNot(_.isEmpty)
 
-  def allPenNames(model: Model): Seq[String] =
-    model.plots.flatMap(_.pens).collect {
-      case pen: Pen => pen
-    }.map(_.display)
+  def allKeyedPenNames(model: Model): Seq[(String, Seq[String])] = {
+    model.plots
+      .map((plot: Plot) => (plot.display.getOrElse(""), plot.pens.toSeq.map(_.display)))
+      .filterNot(_._1.isEmpty)
+  }
 
-  private def determineRenames(names: Seq[String]): Seq[(String, String)] = {
-    names.groupBy(_.toUpperCase)
-      .toSeq
+  private[fileformat] def mapRenameDecision(groupedNames: Seq[(String, Seq[String])]):  Seq[(String, String)] = {
+    groupedNames
       .map {
-        case (upper, originals) => upper -> originals.distinct
+        case (upper, originals) => upper -> originals
       }
       .flatMap {
         case (upper, originals) if originals.length > 1 =>
           // Put lower-case names before uppercase names
           val sortedOriginals = originals.sorted.reverse
           sortedOriginals.zipWithIndex.map {
             case (original, index) if index == 0 => (original, original)
-            case (original, index) => (original, s"${original}_${index}")
+            case (original, index) => {
+              var new_name = s"${original}_${index}"
+              var new_index = index
+              while(groupedNames.map(_._1).contains(new_name.toUpperCase)){
+                new_index = new_index + 1
+                new_name  = s"${original}_${new_index}"
+              }
+              (original, s"${original}_${new_index}")
+            }
           }
         case _ => Seq.empty[(String, String)]
       }
   }
 
+  private[fileformat] def determineMapRenames(names: Seq[(String, Seq[String])]): Seq[(String, Seq[(String, String)])] = {
```

Comment:
The names for this and `determineRenames` are slightly confusing to me - would `determinePenRenames` and `determinePlotRenames` be any better?